src/plugins/dataflow-tool/components/dataflow-tool.tsx

```diff
@@ -23,15 +27,22 @@ interface IProps extends IToolTileProps{
 }
 
 interface IState {
+  isEditingTitle?: boolean;
 }
 
+let nextDataflowProgramId = 0;
```

Comment:
I wasn't able to tell how this was being used, but I think it's worth pointing out that this will change based on the order in which the dataflow tiles are rendered. So if I create two DF tiles (1 and 2), swap the order they appear in the document, then reload the page, I expect their `programId`s will switch. I think there are other cases where these will be inconsistent, such as when multiple documents that include DF tiles are loaded.

Reply:
Thanks, you are absolutely correct.  I wrote an in-depth comment above so as not to forget the path that led there.  Regardless, I've taken it out.